compiler/noirc_errors/src/debug_info.rs

```diff
@@ -27,7 +27,7 @@ pub struct DebugInfo {
     /// Serde does not support mapping keys being enums for json, so we indicate
     /// that they should be serialized to/from strings.
     #[serde_as(as = "BTreeMap<DisplayFromStr, _>")]
-    pub locations: BTreeMap<OpcodeLocation, Vec<Location>>,
+    pub locations: BTreeMap<OpcodeLocation, Vec<Span>>,
```

Comment:
Before merging I think a rationale of this change would be useful. A span and a location are different things

Reply:
They do carry same data: 
Location is Span (offsets) + File/SrcId, 
Span is offsets + context (fileid/srcid),

Not attached one way or the other - it follows however concept defined by library.

Reply:
I think @kevaundray here was likely also confused by the fact a Span is usually just a range, where a location also implies a source file or similar. Chumsky's use of a Span with a context was likely just trying to shoehorn a name into a generic type that could have been either a Span or a Location depending on the context, so they had to choose one.